src/grammar/query.rs

```diff
@@ -1204,6 +1204,153 @@ Root@0..174
+                    ArithmeticOp@93..94 "*"
+                    Whitespace@94..95 " "
+                    Decimal@95..99 "0.05"
+                Comma@99..100 ","
+                Argument@100..101
+                  Integer@100..101 "2"
+              RParen@101..102 ")"
+            Whitespace@102..103 " "
+            Comment@103..109 "-- CPU"
+            Whitespace@109..114 "\n    "
+            Keyword@114..118 "WHEN"
+            Whitespace@118..119 " "
+            ComparissonExpression@119..125
+              Integer@119..120 "2"
+              Whitespace@120..125 "\n    "
+            Keyword@125..129 "THEN"
+            Whitespace@129..130 " "
+            FunctionInvocation@130..155
+              IdentGroup@130..135
+                Ident@130..135 "ROUND"
+              LParen@135..136 "("
+              ArgumentList@136..154
+                Argument@136..152
+                  Expression@136..152
+                    IdentGroup@136..146
+                      Ident@136..146 "List_price"
+                    Whitespace@146..147 " "
+                    ArithmeticOp@147..148 "*"
+                    Whitespace@148..149 " "
+                    Decimal@149..152 "0.1"
+                Comma@152..153 ","
+                Argument@153..154
+                  Integer@153..154 "2"
+              RParen@154..155 ")"
+            Whitespace@155..157 "  "
+            Comment@157..170 "-- Video Card"
+            Whitespace@170..175 "\n    "
+          ElseExpression@175..229
+            Keyword@175..179 "ELSE"
+            Whitespace@179..180 " "
+            FunctionInvocation@180..206
+              IdentGroup@180..185
+                Ident@180..185 "ROUND"
+              LParen@185..186 "("
+              ArgumentList@186..205
+                Argument@186..203
+                  Expression@186..203
+                    IdentGroup@186..196
+                      Ident@186..196 "list_price"
+                    Whitespace@196..197 " "
+                    ArithmeticOp@197..198 "*"
+                    Whitespace@198..199 " "
+                    Decimal@199..203 "0.08"
+                Comma@203..204 ","
+                Argument@204..205
+                  Integer@204..205 "2"
+              RParen@205..206 ")"
+            Whitespace@206..207 " "
+            Comment@207..226 "-- other categories"
+            Whitespace@226..229 "\n  "
+          Keyword@229..232 "END"
+          Whitespace@232..233 " "
+        Alias@233..241
+          Ident@233..241 "discount"
+        Whitespace@241..242 "\n"
+    Keyword@242..246 "FROM"
+    Whitespace@246..249 "\n  "
+    IdentGroup@249..257
+      Ident@249..257 "products"
+    Whitespace@257..258 "\n"
+    OrderByClause@258..281
+      Keyword@258..263 "ORDER"
+      Whitespace@263..264 " "
+      Keyword@264..266 "BY"
+      Whitespace@266..269 "\n  "
+      Expression@269..281
+        IdentGroup@269..281
+          Ident@269..281 "product_name"
+    Semicolon@281..282 ";"
```

Comment:
I don't think this test is necessary, we already know that `parse_expr` can do case statements.